client/token_claim_create.go

```diff
@@ -73,26 +73,26 @@ func (cli *VanClient) TokenClaimCreate(ctx context.Context, name string, passwor
 	}
 	host := fmt.Sprintf("%s.%s", types.ControllerServiceName, cli.Namespace)
 	localOnly := true
-	if cli.RouteClient != nil {
-		route, err := cli.RouteClient.Routes(cli.Namespace).Get(types.ClaimRedemptionRouteName, metav1.GetOptions{})
+	if service.Spec.Type == corev1.ServiceTypeNodePort {
+		host, err = cli.getControllerIngressHost()
 		if err != nil {
 			return err
 		}
-		host = route.Spec.Host
-		port = 443
-		localOnly = false
-	} else if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
-		host = kube.GetLoadBalancerHostOrIp(service)
-		localOnly = false
-	} else if service.Spec.Type == corev1.ServiceTypeNodePort {
-		host, err = cli.getControllerIngressHost()
+		port, err = getClaimsNodePort(service)
 		if err != nil {
 			return err
 		}
-		port, err = getClaimsNodePort(service)
+		localOnly = false
+	} else if cli.RouteClient != nil {
+		route, err := cli.RouteClient.Routes(cli.Namespace).Get(types.ClaimRedemptionRouteName, metav1.GetOptions{})
 		if err != nil {
 			return err
 		}
+		host = route.Spec.Host
+		port = 443
+		localOnly = false
+	} else if service.Spec.Type == corev1.ServiceTypeLoadBalancer {
```

Comment:
Will this not result in the same error if ingress=loadbalancer on openshift? (As previous else if clause will be true).

Reply:
Yes, the approach you recommended is supposed to fix both cases.